### DISCUSSION TITLE
Continuous async requests with HttpClientInterface

### DISCUSSION THREAD

**Discussion by deniz777** [symfony/symfony#42665] [Q&A] [answered]
Hello, I am trying to create async requests with HttpClientInterface in my console app. Please check the code:

```
public function asyncRequest(): void
{
    try {
        $response = $this->client->request('POST', 'url', ['headers' => $headers, 'json' => $data]);
        $content = $response->getContent();
    } catch (\Throwable $th) {
        $this->logger->error($th->getMessage());
    }
}
```
And calling this method in endless loop:
```
while (true) { 
    $this->asyncRequest();
}
```
I don't need response contents but if I don't call `$response->getContent()`, it does not catch the errors and it causes app crashes. If I call, it breaks asynchronization. Do you have any solution for this? I thought creating async processes for functions that needs async requests and let them crash in some other console app but it sounds bad idea.

**Comment by lyrixx** [comment, marked as answer]
Hi, 

Response stays async when it's referenced by a variables and not destructed.
So you need to store all responses somewhere, then check all statuses.

```php
$client = HttpClient::create();

$urls = [
    'https://httpbin.org/status/200',
    'https://httpbin.org/status/500',
    'https://httpbin.org/delay/1',
    'https://httpbin.org/status/404',
];


$responses = [];

foreach ($urls as $url) {
    $responses[] = $client->request('GET', $url);
}

// Do something else while waiting for response

foreach ($responses as $response) {
    if (200 !== $response->getStatusCode()) {
        printf('Oups something where wrong with a request "%s": "%d"' . "\n", $response->getInfo('url'), $response->getStatusCode());
    };
}
```

**Reply by deniz777** [reply]
@lyrixx But when I call `$response->getStatusCode()` it breaks asynchronization, that's my main problem. I want async requests forever, not just for a while and I don't care responses.

**Reply by lyrixx** [reply]
there is something I don't understand 

* you want async
* you don't want to check the status code
* but you want an exception if something is wrong

=> at some point, you must call something to ensure the response is OK...

**Reply by deniz777** [reply]
Actually I don't want exceptions too, I was just trying to keep the app not crashed. So if you say I must check responses, better I try it by creating another processes.

**Reply by lyrixx** [reply]
What do you mean by "the app is crashing" ?
Because if you take my example, remove the manual check (and update all url to be sure to got a 200) everything works fine

**Reply by lyrixx** [reply]
@deniz777 Don't forget to mark the question as resolved if you think it's ok :)

**Reply by deniz777** [reply]
> What do you mean by "the app is crashing" ?
> Because if you take my example, remove the manual check (and update all url to be sure to got a 200) everything works fine

Remove manual check? If I remove `200 !== $response->getStatusCode()` condition, app will throw error when I send invalid request. There will be no catch mechanism and app will stop working. That's what I mean by saying "app crash".

**Reply by deniz777** [reply]
I also tried to mark your comment as answer but could not make it work. There must be some problem with github.
